### PR TITLE
[WIP] Add more operations for torch

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
In this PR I am trying to cover more operations for the torch framework and also add tests as operations are added.
The ideal scenario is to test each operation using all types and check the returned result.
 ```quant``` and ```complex``` can not be tested for the moment.
 * ```complex``` not implemented
 * ```quant``` tensors are created using a different method and not with the constructor

## Status
-  [x] - Add basic operations (add, sub, mul) - tested on all the tensors
-  [ ] - Check why div does not work

### CODE REVIEW

File: tests/syft/lib/torch/tensor_test.py [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
+
+from syft.core.pointer.pointer import Pointer
+from syft.lib.torch.tensor_util import TORCH_STR_DTYPE
+
 import syft as sy
 import torch as th
 import pytest
+from itertools import product
 
+TYPES_TEST = [e for e in TORCH_STR_DTYPE.keys() if not e.startswith("complex")]
```
Comment (gmuraru) [inline tests/syft/lib/torch/tensor_test.py:14]: Need to remove the complex type because is still WIP in torch

Comment (gmuraru) [inline tests/syft/lib/torch/tensor_test.py:14]: Q: @tudorcebere @iamtrask do we want to test all types of tensors that we can send and do operations with them?

Comment (tudorcebere) [reply tests/syft/lib/torch/tensor_test.py:14]: Would be really cool!

Comment (iamtrask) [reply tests/syft/lib/torch/tensor_test.py:14]: Absolutely

Comment (iamtrask) [reply tests/syft/lib/torch/tensor_test.py:14]: Ideally we'd also test them in the context of backpropagation if we can.

**Pull request merged**